tests/unit/test_ecies.cpp

```diff
@@ -0,0 +1,382 @@
+}
+
+TEST(ECIESBuilderTest, incompleteBuilder2)
+{
+    ECIESCtxBuilder decBuilder;
+    std::function<std::unique_ptr<MessageAuthenticationCode>(const std::vector<uint8_t>&)> macFunc =
+            [] (const std::vector<uint8_t> &key) { (void) key; return nullptr; };
+    decBuilder.setMacFactoryFunction(macFunc);
+    EXPECT_THROW(decBuilder.buildDecryptionCtx(randomKey), MoCOCrWException);
+}
+
+TEST(ECIESBuilderTest, incompleteBuilder3)
+{
+    ECIESCtxBuilder decBuilder;
+    decBuilder.setMacKeySize(1);
+    EXPECT_THROW(decBuilder.buildDecryptionCtx(randomKey), MoCOCrWException);
+}
+
+TEST(ECIESBuilderTest, incompleteBuilder4)
+{
+    ECIESCtxBuilder decBuilder;
+    std::function<std::unique_ptr<SymmetricCipherI>(const std::vector<uint8_t>&)> symCipherFunc  =
+                [] (const std::vector<uint8_t> &key) { (void) key; return nullptr; };
+    decBuilder.setSymmetricCipherFactoryFunction(symCipherFunc);
+    EXPECT_THROW(decBuilder.buildDecryptionCtx(randomKey), MoCOCrWException);
+}
+
+TEST(ECIESBuilderTest, incompleteBuilder5)
+{
+    ECIESCtxBuilder decBuilder;
+    decBuilder.setSymmetricCipherKeySize(1);
+    EXPECT_THROW(decBuilder.buildDecryptionCtx(randomKey), MoCOCrWException);
+}
+
+TEST(ECIESTest, certTest)
+{
+    std::string privKeyString{R"(-----BEGIN EC PRIVATE KEY-----
+MHcCAQEEIOGOxWwDXOKW482FT+jFUaS5CQS8iadU78f60nBN3zTUoAoGCCqGSM49
+AwEHoUQDQgAEYBZUUP11bhdM/eNWzSLD1MDTBmEgWkJDnIR4UfKvo2mgShST4U1S
+TcJF1xBUhF5ecYo7g/Niwpw8SwG1QH0a+A==
+-----END EC PRIVATE KEY-----)"
+    };
+
+    std::string certString{R"(-----BEGIN CERTIFICATE-----
+MIICcTCCAhegAwIBAgIURdMbES7SekKg3835htjY+nay3kYwCgYIKoZIzj0EAwIw
+gY0xCzAJBgNVBAYTAkRFMQswCQYDVQQIDAJCVzEMMAoGA1UEBwwDVWxtMRIwEAYD
+VQQKDAlCTVcgQ2FySVQxDTALBgNVBAsMBEpDLTcxETAPBgNVBAMMCERldi1UZWFt
+MS0wKwYJKoZIhvcNAQkBFh5NR1UtU2VjdXJpdHktVGVhbUBsaXN0LmJtdy5jb20w
+HhcNMjAwNDE0MTY1NjMxWhcNMjEwNDE0MTY1NjMxWjCBjTELMAkGA1UEBhMCREUx
+CzAJBgNVBAgMAkJXMQwwCgYDVQQHDANVbG0xEjAQBgNVBAoMCUJNVyBDYXJJVDEN
+MAsGA1UECwwESkMtNzERMA8GA1UEAwwIRGV2LVRlYW0xLTArBgkqhkiG9w0BCQEW
+Hk1HVS1TZWN1cml0eS1UZWFtQGxpc3QuYm13LmNvbTBZMBMGByqGSM49AgEGCCqG
+SM49AwEHA0IABGAWVFD9dW4XTP3jVs0iw9TA0wZhIFpCQ5yEeFHyr6NpoEoUk+FN
+Uk3CRdcQVIReXnGKO4PzYsKcPEsBtUB9GvijUzBRMB0GA1UdDgQWBBSi+rthnvaF
+Hd/ceL/FukSRixvuAzAfBgNVHSMEGDAWgBSi+rthnvaFHd/ceL/FukSRixvuAzAP
+BgNVHRMBAf8EBTADAQH/MAoGCCqGSM49BAMCA0gAMEUCIF/JBFRAROnQAqmpcs5f
++Q8+M51Rj0qS5X8+7d/eWfEXAiEAzih37HPbeFvoP36Xqh4GpXfsU1BZp9RB4Q7S
+VndFGPg=
+-----END CERTIFICATE-----)"
+    };
+
+    auto privKey(AsymmetricKeypair::readPrivateKeyFromPEM(privKeyString, ""));
+    auto cert = X509Certificate::fromPEM(certString);
+
+    ECIESCtxBuilder encBuilder;
+    ECIESCtxBuilder decBuilder;
+
+    const std::vector<uint8_t> zeroIV = std::vector<uint8_t>(AESCipherBuilder::getDefaultIVLength(mode));
+
+    auto kdf = std::make_shared<X963KDF>(openssl::DigestTypes::SHA512);
+
+    auto cipherDecFunc = [zeroIV](const std::vector<uint8_t> &key) -> std::unique_ptr<AESCipher> {
```

Comment:
Wouldn't the defaults for all of these parameters be sufficient in this test?

Reply:
Done